darkcluster/src/main/java/com/linkedin/darkcluster/impl/DarkClusterManagerImpl.java

```diff
@@ -54,19 +54,27 @@
   private final Facilities _facilities;
   private final String _sourceClusterName;
   private final DarkClusterStrategyFactory _darkClusterStrategyFactory;
+  private final DarkDispatcherVerifier _darkDispatcherVerifier;
   private Map<String, AtomicReference<URIRewriter>> _uriRewriterMap;
 
   public DarkClusterManagerImpl(@Nonnull String sourceClusterName, @Nonnull Facilities facilities,
                                 @Nonnull DarkClusterStrategyFactory strategyFactory, String whiteListRegEx,
                                 String blackListRegEx, @Nonnull Notifier notifier)
   {
+    this(sourceClusterName, facilities, strategyFactory, whiteListRegEx, blackListRegEx, notifier, null);
+  }
+
+  public DarkClusterManagerImpl(@Nonnull String sourceClusterName, @Nonnull Facilities facilities,
+      @Nonnull DarkClusterStrategyFactory strategyFactory, String whiteListRegEx,
+      String blackListRegEx, @Nonnull Notifier notifier, DarkDispatcherVerifier darkDispatcherVerifier) {
     _whiteListRegEx = whiteListRegEx == null ? null : Pattern.compile(whiteListRegEx);
     _blackListRegEx = blackListRegEx == null ? null : Pattern.compile(blackListRegEx);
     _notifier = notifier;
     _facilities = facilities;
     _sourceClusterName = sourceClusterName;
     _darkClusterStrategyFactory = strategyFactory;
     _uriRewriterMap = new HashMap<>();
+    _darkDispatcherVerifier = darkDispatcherVerifier;
```

Comment:
lets  create a no op implementation and do the null check here, instead of on every request.